File: abc_spec/code_samples/C#/instruments@{id}/patch.cs
// For more information please refer to https://github.com/checkout/checkout-sdk-net
using Checkout.Common;
using Checkout.Instruments;

ICheckoutApi api = CheckoutSdk.DefaultSdk().StaticKeys()
    .PublicKey("public_key")
    .SecretKey("secret_key")
    .Environment(Environment.Sandbox)
    .HttpClientFactory(new DefaultHttpClientFactory())
    .Build();

UpdateInstrumentRequest request = new UpdateInstrumentRequest
{
    ExpiryMonth = 10,
    ExpiryYear = 2027,
    Name = "FirstName LastName",
    AccountHolder = new InstrumentAccountHolder()
    {
        BillingAddress = new Address()
        {
            AddressLine1 = "Checkout.com",
            AddressLine2 = "90 Tottenham Court Road",
            City = "London",
            State = "London",
            Zip = "W1T 4TJ",
            Country = CountryCode.GB
        },
        Phone = new Phone()
        {
            Number = "4155552671",
            CountryCode = "1"
        },
    },
    Customer = new UpdateInstrumentRequest.UpdateInstrumentCustomer()
    {
        Id = "cus_gajmdgunwwlehbctuj6a3sifpm",
        Default = true
    }
};

try
{
    UpdateInstrumentResponse response = api.InstrumentsClient().Update("instrument_id", request).Result;
}
catch (CheckoutApiException e)
{
    // API error
    string requestId = e.RequestId;
    var statusCode = e.HttpStatusCode;
    IDictionary<string, object> errorDetails = e.ErrorDetails;
}
catch (CheckoutArgumentException e)
{
    // Bad arguments
}
catch (CheckoutAuthorizationException e)
{
    // Invalid authorization
}

File: abc_spec/code_samples/PHP/instruments@{id}/patch.php
<?php
//For more information please refer to https://github.com/checkout/checkout-sdk-php

use Checkout\\CheckoutApiException;
use Checkout\\CheckoutAuthorizationException;
use Checkout\\CheckoutSdk;
use Checkout\\Common\\Address;
use Checkout\\Common\\Country;
use Checkout\\Common\\Phone;
use Checkout\\Environment;
use Checkout\\Instruments\\Previous\\InstrumentAccountHolder;
use Checkout\\Instruments\\Previous\\UpdateInstrumentRequest;

$api = CheckoutSdk::builder()
    ->previous()
    ->staticKeys()
    ->environment(Environment::sandbox())
    ->secretKey("secret_key")
    ->build();

$address = new Address();
$address->address_line1 = "CheckoutSdk.com";
$address->address_line2 = "90 Tottenham Court Road";
$address->city = "London";
$address->state = "London";
$address->zip = "W1T 4TJ";
$address->country = Country::$GB;

$phone = new Phone();
$phone->country_code = "+1";
$phone->number = "415 555 2671";

$instrumentAccountHolder = new InstrumentAccountHolder();
$instrumentAccountHolder->billing_address = $address;
$instrumentAccountHolder->phone = $phone;

$request = new UpdateInstrumentRequest();
$request->expiry_month = 10;
$request->expiry_year = 2027;
$request->name = "FirstName LastName";
$request->account_holder = $instrumentAccountHolder;

try {
    $response = $api->getInstrumentsClient()->update("instrument_id", $request);
} catch (CheckoutApiException $e) {
    // API error
    $error_details = $e->error_details;
    $http_status_code = isset($e->http_metadata) ? $e->http_metadata->getStatusCode() : null;
} catch (CheckoutAuthorizationException $e) {
    // Bad Invalid authorization
}
